### PR TITLE
feat(focuslock): add focus traps

### PR DESCRIPTION
**Pull request by jamesb3ll**
Focus Traps have been added for cases where the focus lock element is rendered at the beginning or end of the DOM. If `[TAB]` key is pressed when the currently focused element is the last focusable element in the DOM tree, the browser will focus outside the document and not fire the `focusin` event listener. To fix this I've added some focus traps surrounding the focus lock container, these traps have `tab-index="0"` to force a focusable state, if one of these is focused, the focus will be reset to inside the locked area.

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/184)

### CODE REVIEW

File: src/components/FocusLock/index.js [hunk 1]
```diff
@@ -62,16 +69,23 @@ export default function FocusLock(props) {
...
   }, [active]);
 
-  return <Container ref={baseRef}>{children}</Container>;
+  return (
+    <>
+      <span tabIndex={0} ref={topRef} />
+      <Container ref={containerRef} {...otherProps}>
+        {children}
+      </Container>
+      <span tabIndex={0} ref={bottomRef} />
```
Comment (congcongcong250) [inline src/components/FocusLock/index.js]: nice trap

File: src/components/FocusLock/index.js [hunk 2]
```diff
@@ -24,19 +25,25 @@ const getTabbables = el =>
...
+  const bottomRef = useRef();
 
   useLayoutEffect(() => {
     const onFocus = event => {
-      // Focused element is _outside_ our component
-      if (active && !baseRef.current.contains(event.target)) {
-        const tabbables = getTabbables(baseRef.current);
+      if (
+        active &&
+        (!containerRef.current.contains(event.target) ||
```
Comment (congcongcong250) [inline src/components/FocusLock/index.js]: Potential bug cases could be some other part triggering `.focus()` and it magically focus onto the first element`tabbables[0]` in this Container. 
**Just FYI. No need to think of solutions unless the bug were actually happening.**

Comment (jamesb3ll) [reply src/components/FocusLock/index.js]: Sure, if `.focus()` is called on an element outside the container this code will re-focus the first tabbable element inside the container. I think this is a feature 😄

**Pull request merged**